### ISSUE TITLE
Consider leaving SA1118 (ParameterMustNotSpanMultipleLines) disabled by default

### ISSUE THREAD

**Issue by sharwell** [DotNetAnalyzers/StyleCopAnalyzers#823]
My experience with SA1118 in multiple projects is it rarely (never so far) reports warnings in locations that need to be corrected for readability.

In light of other parameter readability rules (especially SA1117 ParametersMustBeOnSameLineOrSeparateLines), I believe the product will be improved by leaving SA1118 disabled by default in the official distribution.

**Comment by sharwell** [comment]
If #748 was accepted, SA1118 would provide *even less* value, because the operator placement at the beginning of a line would make it immediately obvious that the following call has three arguments, not four.

```csharp
SomeMethod(
    Value1,
    ValueWithVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongName
        + ValueWithVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongName,
    Value3);
```

**Comment by jackwhelpton** [comment]
I've found this rule relatively helpful in previous releases of StyleCop, as it would push us towards:

```csharp
var value2 = ValueWithVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongName
           + ValueWithVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryVeryLongName;

SomeMethod(
    Value1,
    value2,
    Value3);
```

which minimised context-switching (setup parameters, call, rather than call (oh and in the process set up parameters)).

As I mentioned in https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/818, the main false positive I'm seeing relative to the previous version of StyleCop is that it seemed to adopt the rule:

> Each parameter to a method must either:
> 
> * Fit on a single line
> * Be an anonymous method
> * **Be _itself_ a method call, in which case the same rules apply to _its_ parameters**

I can't see the third rule in the official documentation, but it would explain why code like the following previously passed the requirements of SA1118:

```csharp
var t = new Test(
    new Test(
        new Test()),
    new Test(
        new Test()));
```

Would this be a big change? If time allows I could have a look at a pull request, although I'd be approaching Roslyn with no prior experience.

On an unrelated note, I always found the exception for the first parameter in SA1118 rather odd. It might make more sense if it was an exception for methods with only one parameter, although even then I'd prefer the intermediate variable approach.

**Comment by sharwell** [comment]
@jackwhelpton It sounds like StyleCop Classic actually implemented the exception I proposed in #1597. Based on that information, I'll mark #1597 as a blocking enhancement for RC 1. I'm also marking this issue as a duplicate of that one, even though it came later.
